main.tf

```diff
@@ -100,14 +100,14 @@ resource "azurerm_container_registry" "acr" {
   # Moving from deprecated argument, georeplication_locations, but keeping container_registry_geo_replica_locs
   # for backwards compatability.
   #
-  georeplications = (local.container_registry_sku == "Premium" && var.container_registry_geo_replica_locs != null) ? [
-    for location_item in var.container_registry_geo_replica_locs:
-      {
-        location = location_item
-        tags     = var.tags
-      }
-  ] : local.container_registry_sku == "Premium" ? [] : null
-
+  dynamic "georeplications" {
+    for_each = (local.container_registry_sku == "Premium" && var.container_registry_geo_replica_locs != null) ? toset(
+      var.container_registry_geo_replica_locs) : []
+    content {
+      location                = georeplications.key
+      tags                    = var.tags
+    }
+  }
```

Comment:
Was there a need to change this to a dynamic block? Asking in general as the new module also treats the `georeplications` parameter as optional and could take a value of `null`. Not saying the dynamic block is a bad thing, just wondering why we spent the time to re-write the code here.

Reply:
The argument "georeplications" is updated to be defined as a block of type. You are right, the new module does treat `georeplications` as an optional argument and can accept null to ignore the block from being added.
The original code had logic in place for `container_registry_sku = "Premium" and container_registry_geo_replica_locs is set` so updated the existing property to a dynamic block to support the case were user provides multiple geo replication locations. And setting empty `[]`, omits the property

Reply:
Sounds good.